exporters/deploytime/app.py

```diff
@@ -50,54 +52,74 @@ def image_sha(img_url):
-        replicationcontrollers = v1_replicationcontrollers.get(namespace=namespace,
-                                                               label_selector=pelorus.get_app_label())
-        replicas = replicationcontrollers.items
-
-        # Process ReplicaSets from apps/v1 api version for Deployments
-        v1_replicationsets = dyn_client.resources.get(api_version='apps/v1', kind='ReplicaSet')
-        replicationsets = v1_replicationsets.get(namespace=namespace,
-                                                 label_selector=pelorus.get_app_label())
-        replicas = replicas + replicationsets.items
-
-        # Process ReplicaSets from extentions/v1beta1 api version for Deployments
-        v1_replicationsets = dyn_client.resources.get(api_version='extensions/v1beta1', kind='ReplicaSet')
-        replicationsets = v1_replicationsets.get(namespace=namespace,
-                                                 label_selector=pelorus.get_app_label())
-        replicas = replicas + replicationsets.items
-
-        for rc in replicas:
-            images = [image_sha(c.image) for c in rc.spec.template.spec.containers]
-            print("IMAGE" + str(images))
-
-            # Since a commit will be built into a particular image and there could be multiple
-            # containers (images) per pod, we will push one metric per image/container in the
-            # pod template
-            for i in images:
-                if i is not None:
-                    metric = DeployTimeMetric(rc.metadata.name, namespace)
-                    metric.labels = rc.metadata.labels
-                    metric.deploy_time = rc.metadata.creationTimestamp
-                    metric.image_sha = i
-                    metrics.append(metric)
+        logging.info("Wwtching namespaces %s", namespaces)
+
+    pods = v1_pods.get(label_selector=pelorus.get_app_label(), field_selector='status.phase=Running').items
+
+    replicas_dict = {}
+    # Process ReplicationControllers for DeploymentConfigs
+    get_replicas('v1', 'ReplicationController', replicas_dict)
+
+    # Process ReplicaSets from apps/v1 api version for Deployments
+    get_replicas('apps/v1', 'ReplicaSet', replicas_dict)
+
+    # Process ReplicaSets from extentions/v1beta1 api version for Deployments
+    get_replicas('extensions/v1beta1', 'ReplicaSet', replicas_dict)
+
+    for pod in pods:
+        if not namespaces or (pod.metadata.namespace in namespaces):
+            logging.info("Getting Replicas for pod: %s", pod.metadata.name)
+            ownerRefs = pod.metadata.ownerReferences
+            namespace = pod.metadata.namespace
+
+            for ownerRef in ownerRefs:
+                if ownerRef.kind in ['ReplicaSet', 'ReplicationController'] and not pod_replica_dict.get(ownerRef.name):
+
+                    logging.info("Getting Replica ownerRef: %s, %s",  ownerRef.name, ownerRef.kind)
```

Comment:
Same as above. Switch to Debug level log, and include the namespace in the message.